File: function-aws-alexa/features/function-aws-alexa-java/skeleton/src/main/java/@defaultPackage.path@/Application.java
package @defaultPackage@;

import javax.inject.Singleton;
import java.util.Optional;
import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.model.Response;
import io.micronaut.function.aws.alexa.annotation.IntentHandler;

@Singleton
public class Application {

    @IntentHandler("ExampleIntent") 
    public Optional<Response> exampleHandler(HandlerInput input) { 
        String speechText = "Example Response";
        return input.getResponseBuilder()
                .withSpeech(speechText)
                .withSimpleCard("ExampleText", speechText)
                .build();
    }    
}

File: function-aws-alexa/features/function-aws-alexa-kotlin/skeleton/src/main/kotlin/@defaultPackage.path@/Application.kt
package @defaultPackage@

import javax.inject.Singleton
import java.util.Optional
import com.amazon.ask.dispatcher.request.handler.HandlerInput
import com.amazon.ask.model.Response
import io.micronaut.function.aws.alexa.annotation.IntentHandler

@Singleton
class Application {

    @IntentHandler("ExampleIntent") 
    fun exampleHandler(input : HandlerInput) : Optional<Response> { 
        val speechText = "Example Response"
        return input.responseBuilder
                .withSpeech(speechText)
                .withSimpleCard("ExampleText", speechText)
                .build()
    }    
}
